lib/async/redis/sentinels.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module Async
+	module Redis
+		class SentinelsClient < Client
+			def initialize(master_name, sentinels, role = :master, protocol = Protocol::RESP2, **options)
+				@master_name = master_name
+				@sentinel_endpoints = sentinels.map do |sentinel|
+					Async::IO::Endpoint.tcp(sentinel[:host], sentinel[:port])
+				end
+				@role = role
+
+				@protocol = protocol
+				@pool = connect(**options)
+			end
+
+			private
+
+			# Override the parent method. The only difference is that this one needs
+			# to resolve the master/slave address.
+			def connect(**options)
+				Async::Pool::Controller.wrap(**options) do
+					addr = resolve_addr
+					endpoint = Async::IO::Endpoint.tcp(addr[:ip], addr[:port])
+
+					peer = endpoint.connect
+
+					peer.setsockopt(Socket::IPPROTO_TCP, Socket::TCP_NODELAY, 1)
```

Comment:
Maybe my fault, but I don't think this line is needed any more. I think `IO::Stream.new(peer) should be sufficient.

https://github.com/socketry/async-io/blob/1aec9882581aaf84e702e01afca6a54a00ffb08e/lib/async/io/stream.rb#L56

Then

https://github.com/socketry/async-io/blob/156fa95d79bd058d92e2555a81fd3807648f1377/lib/async/io/peer.rb#L54-L69

Reply:
I've deleted it and it seems that it works :+1: